### PR TITLE
[wip] Replace calls to ProposalObject with Proposal

### PR DESCRIPTION
**Pull request by bkutil**
This patchset replaces most of the calls to ProposalObject with calls to Proposal in the crowbar barclamp.

What has not been replaced and is still backed by databags:

  * The deployment queue
  * Network configuration

### CODE REVIEW

**Comment by bkutil** [comment]
Tested by patching an admin node and applying / saving / modifying the NFS barclamp.

File: crowbar_framework/app/models/service_object.rb [hunk 1]
```diff
@@ -604,17 +604,17 @@ def element_info(role = nil)
...
 
   def proposals_raw
-    ProposalObject.find_proposals(@bc_name)
+    Proposal.where(barclamp: @bc_name)
   end
 
   def proposals
     props = proposals_raw
-    props.map! { |p| p["id"].gsub("bc-#{@bc_name}-", "") } unless props.empty?
+    props = props.map { |p| p["id"].gsub("bc-#{@bc_name}-", "") } unless props.empty?
```
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: I know it's not your fault, but maybe we can drop the useless `unless` here?

**Comment by vuntz** [comment]
Need to look at it tomorrow, but after a quick glance, looks good.

**Comment by bkutil** [comment]
@vuntz I removed the extra `unless x.empty?`

File: crowbar_framework/app/controllers/barclamp_controller.rb [hunk 2]
```diff
@@ -369,9 +367,8 @@ def proposal_update
       return render :text => ret[1], :status => ret[0] if ret[0] != 200
       return render :json => ret[1]
     else # This is UI.
-      params[:id] = "bc-#{params[:barclamp]}-#{params[:id] || params[:name]}"
       if params[:submit] == t('barclamp.proposal_show.save_proposal')
-        @proposal = ProposalObject.find_proposal_by_id(params[:id])
+        @proposal = Proposal.where(barclamp: params[:barclamp], name: params[:id] || params[:name]).first
```
Comment (vuntz) [inline crowbar_framework/app/controllers/barclamp_controller.rb]: I'm a bit unsure here: it seems `params[:name]` would be the full name (`bc-barclamp-foo`), in which case I don't think it's correct to use it that way.

File: crowbar_framework/app/controllers/barclamp_controller.rb [hunk 3]
```diff
@@ -382,7 +379,7 @@ def proposal_update
           flash_and_log_exception(e)
         end
       elsif params[:submit] == t('barclamp.proposal_show.commit_proposal')
-        @proposal = ProposalObject.find_proposal_by_id(params[:id])
+        @proposal = Proposal.where(barclamp: params[:barclamp], name: params[:id] || params[:name]).first
```
Comment (vuntz) [inline crowbar_framework/app/controllers/barclamp_controller.rb]: (same comment)

**Comment by vuntz** [comment]
@bkutil did you also test creating a new proposal with the patch?

File: crowbar_framework/app/controllers/barclamp_controller.rb (hunk 2 above)
Comment (bkutil) [inline crowbar_framework/app/controllers/barclamp_controller.rb]: Looking at how the `params[:id]` is constructed on line [372](https://github.com/crowbar/barclamp-crowbar/pull/1264/files#diff-6e0801372d9a358289edc702ef124633L372) - it seems that this should continue working as expected?

**Comment by bkutil** [comment]
@vuntz yes. I created a new NFS proposal, then added nodes, saved, then deployed. I'll try to test the whole admin node installation process today together w/ Bernhard.

File: crowbar_framework/app/controllers/barclamp_controller.rb (hunk 2 above)
Comment (vuntz) [inline crowbar_framework/app/controllers/barclamp_controller.rb]: Doh, sorry, I had misread the code :-)

**Comment by vuntz** [comment]
+1

File: crowbar_framework/app/controllers/barclamp_controller.rb (hunk 2 above)
Comment (bkutil) [inline crowbar_framework/app/controllers/barclamp_controller.rb]: No worries, it is indeed weird...

**Comment by MaximilianMeister** [comment]
+1

cool, didn't know about `with_indifferent_access` yet.

**Comment by mjura** [comment]
+1

**Pull request merged**